src/toHaveStyleRule.js

```diff
@@ -68,7 +68,7 @@ const getModifiedClassName = (className, staticClassName, modifier = "") => {
 };
 
 const hasClassNames = (classNames, selectors, options) => {
-  const staticClassNames = classNames.filter(x => x.startsWith("sc-"));
+  const staticClassNames = classNames.filter(x => /^(\w+-)?sc-\w+/.test(x));
```

Comment:
I don't think we want the `\w+` part at the end

Reply:
cc @vxcamiloxv

Reply:
done https://github.com/styled-components/jest-styled-components/pull/302/commits/002ccc5373bdc021145555fd4d3b04a2612bc5bc cc @probablyup